cmd/minishift/cmd/start.go

```diff
@@ -307,12 +308,18 @@ func clusterUp(config *cluster.MachineConfig) {
 		OpenShiftVersion:  config.OpenShiftVersion,
 		MinishiftCacheDir: filepath.Join(constants.Minipath, "cache"),
 	}
-	err := oc.EnsureIsCached()
-	if err != nil {
+	if err := oc.EnsureIsCached(); err != nil {
 		glog.Errorln("Error starting the cluster: ", err)
 		atexit.Exit(1)
 	}
 
+	// Update MACHINE_NAME.json for oc path
+	instanceConfig.Config.OcPath = filepath.Join(oc.GetCacheFilepath(), constants.OC_BINARY_NAME)
+	if err := instanceConfig.Config.WriteInstanceConfig(); err != nil {
```

Comment:
So why _InstanceConfig_, that's really redundant, right? A simple _Write_ should do. WDYT?